### PR TITLE
fix: bump/clean up template deps

### PR DESCRIPTION
**Pull request by cristiand391**
Bumps template deps to latest, remove salesforce ones, add bin/dev.

@W-10996497@

### CODE REVIEW

File: src/generators/app.ts [hunk 1]
```diff
@@ -275,47 +271,34 @@ class App extends Generator {
...
-      'pretty-quick': '^3',
       'ts-node': '^10',
       typescript: '4',
-      mocha: '^8',
+      mocha: '^9',
       nyc: '^15',
       chai: '^4',
       sinon: '10.0.0',
-      '@oclif/test': '^1',
+      '@salesforce/ts-sinon': '^1',
```
Comment (cristiand391) [inline src/generators/app.ts:297]: this is needed by `salesforce/command` when running UTs.

File: src/generators/app.ts [hunk 2]
```diff
@@ -275,47 +271,34 @@ class App extends Generator {
...
       '@typescript-eslint/parser': '^4',
       'eslint-config-prettier': '^8',
-      'eslint-config-salesforce': '^0',
-      'eslint-config-salesforce-typescript': '^0',
       'eslint-plugin-header': '^3',
       'eslint-plugin-import': '^2',
       'eslint-plugin-jsdoc': '^35',
       'eslint-plugin-prettier': '^3',
       'eslint-plugin-typescript': '^0',
-      husky: '^4',
```
Comment (cristiand391) [inline src/generators/app.ts:305]: husky wasn't being used.

File: templates/sfdxPlugin/tsconfig.json [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
-    "rootDir": "src"
+    "rootDir": "src",
+    "module": "commonjs",
+    "target": "es2020",
+    "lib": ["es2020"],
+    "sourceMap": true,
+    "declaration": true,
+    "moduleResolution": "node",
+    "alwaysStrict": true,
+    "noUnusedLocals": true
```
Comment (cristiand391) [inline templates/sfdxPlugin/tsconfig.json:12]: dev-config/tsconfig defaults: 
https://github.com/forcedotcom/dev-config/blob/25549b0ee39c22e44f0c3c627b1da03635d55b17/tsconfig.json#L1

**Review by WillieRuemmele — APPROVED** [review]

**Comment by WillieRuemmele** [comment]
**QA Notes**
---
in `package.json`
```diff
-    "@oclif/command@^1",
-    "@oclif/config@^1",
-    "@oclif/dev-cli@^1",
-    "@oclif/errors@^1",
-    "@oclif/plugin-help@^3",
-    "@oclif/test@^1",
-    "@salesforce/command@^4",
-    "@salesforce/core@^2",
-    "@salesforce/dev-config@^2",
-    "@salesforce/dev-scripts@^0",
-    "@salesforce/prettier-config@^0",
+    "@oclif/core@^1",
+    "@oclif/test@^2",
+    "@salesforce/command@^5",
+    "@salesforce/core@^3",
```

**Pull request merged**